File: tests/testdata/codegen/synthetic-resource-properties-pp/python/synthetic-resource-properties.py
import khulnasoft
import khulnasoft_synthetic as synthetic

rt = synthetic.resource_properties.Root("rt")
khulnasoft.export("trivial", rt)
khulnasoft.export("simple", rt.res1)
khulnasoft.export("foo", rt.res1.obj1.res2.obj2)
khulnasoft.export("complex", rt.res1.obj1.res2.obj2.answer)

File: tests/testdata/codegen/synthetic-resource-properties-pp/nodejs/synthetic-resource-properties.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as synthetic from "@khulnasoft/synthetic";

const rt = new synthetic.resourceproperties.Root("rt", {});
export const trivial = rt;
export const simple = rt.res1;
export const foo = rt.res1.apply(res1 => res1.obj1?.res2?.obj2);
export const complex = rt.res1.apply(res1 => res1.obj1?.res2?.obj2?.answer);
